### PR TITLE
Add MPS merge rules

### PR DESCRIPTION
**Pull request by malfet**
This enables MPS PR reviewed by @kulinseth and @razarmehr to be merged into default branch using mergebot.
"Lint" is the only mandatory check for this rule, as MacOS and iOS  builds are on trunk-only for now

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Thanks!

File: .github/merge_rules.json [hunk 1]
```diff
@@ -273,6 +273,19 @@
...
+      "name": "MPS",
+      "patterns": [
+         "test/test_mps.py",
+         "aten/src/ATen/mps/**",
+         "aten/src/ATen/native/mps/**"
+      ],
+      "approved_by": ["kulinseth"],
+      "mandatory_checks_name": [
+         "Facebook CLA Check",
+         "Lint"
```
Comment (janeyx99) [inline .github/merge_rules.json:286]: I'm guessing we would want to add an MPS build here once an MPS build can be added to CI?

**Review by atalman — APPROVED** [review]

**Review by janeyx99 — APPROVED** [review]

File: .github/merge_rules.json (hunk 1 above)
Comment (malfet) [reply .github/merge_rules.json:286]: We don't want to force `ciflow/trunk` on all MPS PRs, but they are affecting only MacOS builds

**Review by seemethere — APPROVED** [review]

**Pull request closed without merging**